projects/zopfli/zopfli_compress_fuzzer.cc

```diff
@@ -0,0 +1,29 @@
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <string>
+
+#include "zopfli.h"
+#include "FuzzedDataProvider.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  FuzzedDataProvider stream(data, size);
+  ZopfliOptions options;
+  ZopfliInitOptions(&options);
+
+  const ZopfliFormat format = stream.PickValueInArray(
+    {ZOPFLI_FORMAT_GZIP, ZOPFLI_FORMAT_ZLIB, ZOPFLI_FORMAT_DEFLATE});
+
+  unsigned char* outbuf = nullptr;
+  size_t outsize = 0;
+  std::string input = stream.ConsumeRemainingBytesAsString();
```

Comment:
you can do

```cpp
input = stream.ConsumeRemainingBytes<unsigned char>();
```

and then use `input.data(), input.size()` below without extra casts

Reply:
This was the older version. A PR mistake. The new version does not contain the FuzzedDataProvider.h, which is what the developer insisted.

Reply:
Please leave a comment when the PR is ready for review.